jetstream/cli.py

```diff
@@ -188,7 +208,46 @@ def execute(
         ] = ExternalConfigCollection.from_github_repo,
         today: Optional[datetime] = None,
     ) -> bool:
+        run_experiments = self._experiments_to_analyse(experiment_getter)
+        worklist = []
+
+        for experiment in run_experiments:
+            if self.date == All:
+                today = today or self._today()
+                end_date = min(
+                    experiment.end_date or today,
+                    today,
+                )
+                run_dates = inclusive_date_range(experiment.start_date, end_date)
+            else:
+                run_dates = [self.date]
+
+            for run_date in run_dates:
+                assert experiment.normandy_slug
+                worklist.append((experiment.normandy_slug, run_date))
+
+            if self.recreate_enrollments:
+                self._delete_enrollment_table(experiment)
```

Comment:
Moving the decision whether to delete (and later recreate) enrollment tables into `AnalysisExecutor`